localstack/services/opensearch/provider.py

```diff
@@ -400,6 +406,51 @@ def get_store(region: str = None) -> OpenSearchStore:
             ), "The region was not given and it could not be extracted from a request context."
         return opensearch_stores[get_aws_account_id()][region]
 
+    def accept_state_visitor(self, visitor: StateVisitor):
+        visitor.visit(opensearch_stores)
+        visitor.visit(AssetDirectory(os.path.join(config.dirs.data, "opensearch")))
+        visitor.visit(AssetDirectory(os.path.join(config.dirs.data, "elasticsearch")))
+
+    def on_after_state_load(self):
+        """Starts clusters whose metadata has been restored."""
+        for account_id, region_bundle in opensearch_stores.items():
+            for region, store in region_bundle.items():
+                for domain_name, domain_status in store.opensearch_domains.items():
+                    domain_key = DomainKey(domain_name, region, account_id)
+                    if cluster_manager().get(domain_key.arn):
+                        # cluster already restored in previous call to on_after_state_load
+                        continue
+
+                    LOG.info(f"Restoring domain {domain_name} in region {region}.")
+                    try:
+                        preferred_port = None
+                        if config.OPENSEARCH_ENDPOINT_STRATEGY == "port":
```

Comment:
That's an interesting problem actually that might bite us in a few other places as well when a feature flag changes between different persistence-enabled runs :thinking:

Reply:
indeed, already had to disable ports in rds snapshot tests https://github.com/localstack/localstack-ext/pull/1397/files#diff-9554b233d544d2d605d0a8067d419a1adc3ee3dfff6500f6524605217b7aaf65R7-R9